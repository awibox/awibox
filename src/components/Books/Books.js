import React from 'react';
import PropTypes from 'prop-types';
// Styles
import styles from './Books.scss';

const Books = (props) => (
  <div className={styles.books}>
    {props.data.map((book) => (
      <div key={book.name}>
        <a href={book.link} target="_blank" rel="noopener noreferrer">{book.name}</a>
      </div>
    ))}
  </div>
);

Books.propTypes = {
  data: PropTypes.any,
};

export default Books;
