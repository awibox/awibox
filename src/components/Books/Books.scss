@import '../../styles/variables';

.books {
}

.bookItem {
  padding: 10px 15px;
  border-bottom: 1px solid #e7eaec;
  &:last-child {
    border-bottom: none;
  }
}

.bookName {
  font-size: 18px;
  line-height: 24px;
  text-decoration: none;
  margin-right: 10px;
  color: $green;
  &:hover {
    color: $darkGreen;
  }
}
.author {
  font-size: 12px;
}
