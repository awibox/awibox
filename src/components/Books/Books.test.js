import React from 'react';
import { shallow } from 'enzyme';
import Books from './Books';

const mockData = [
  {
    name: 'Name',
    author: 'Author',
    link: 'link',
  },
];

describe('Books', () => {
  it('should render correctly', () => {
    const component = shallow(<Books data={mockData} />);
    expect(component).toMatchSnapshot();
  });
});
