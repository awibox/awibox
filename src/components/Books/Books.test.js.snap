// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books should render correctly 1`] = `
<div>
  <div
    className="bookItem"
    key="Name"
    onClick={[Function]}
    title="Name"
  >
    <div
      className="bookName"
    >
      Name
    </div>
    <div
      className="author"
    >
      Author: 
      <b>
        Author
      </b>
    </div>
  </div>
</div>
`;
